### ISSUE TITLE
Adding plugin in `included` breaks DAG

### ISSUE THREAD

**Issue by thedeeno** [ember-cli/ember-cli#4173]
If I have a plugin with the following `package.json` excerpt:

```
  "ember-addon": {
    "before": [
      "ember-cli-htmlbars"
    ]
  }
```

If register the plugin in the addon's `included`, ember-cli executes the plugin in the wrong order - despite my `before ember-cli-htmlbars` indication above. More explicitly, this doesn't work as expected:

```
module.exports = {
  name: 'my-plugin',
  included: function(app) {
    app.registry.add('template', new TemplatePlugin());
  },
}
```

Adding the plugin by overriding `setupPreprocessorRegistry` DOES work as expected though:

```
module.exports = {
  name: 'my-plugin',
  setupPreprocessorRegistry: function(type, registry) {
    registry.add('template', new TemplatePlugin());
  },
}
```

Seems like a bug?

**Comment by stefanpenner** [comment]
I believe this is a dupe. Although searching from my phone doesn't reveal anything. Although I recall reporting this myself

Yes it is a real issue

**Comment by rwjblue** [comment]
The correct place to make additions to the registry is in the (appropriately named) `setupPreprocessorRegistry` hook, not `included`.  During the transition period folks would have to detect if `setupPreprocessorRegistry` would be called and if not they would use `included` and might end up with this issue, but 0.2.0 was a loooong time ago and this is almost certainly not an issue any longer.

Closing (happy to reopen if I've got something wrong here though)...
